apps/glusterfs/app_volume.go

```diff
@@ -246,8 +246,22 @@ func (a *App) VolumeDelete(w http.ResponseWriter, r *http.Request) {
 			return err
 		}
 
-		return nil
+		if !volume.Info.Block {
```

Comment:
So there are two volumes 

* Block Volumes
* Block Hosting Volumes(File volumes)

When a Block volume of 1 GB is requested, heketi creates a block hosting volume which will be of size 1TB(default value which is configurable). This block hosting volume will be used by gluster block to create 1 GB volume. The remaining size will be used for future requests, So BlockCreate will check if there are block hosting volume already present and then if there is space in them to allocate block volume requested. If not create a new block hosting volume. This approach is documented in https://github.com/gluster/gluster-kubernetes/pull/268.